src/controllers/AuthenticationController.ts

```diff
@@ -0,0 +1,345 @@
+import {NextFunction, Request, Response} from 'express';
+import userFriendlyMessage from '../consts/userFriendlyMessages';
+import UserService from '../services/UserService';
+import EmailService from '../services/EmailService';
+import JWTUtils from '../utils/jwtUtils';
+import PasswordUtils from '../utils/passwordUtils';
+import {
+  Payload,
+  Role,
+  UserAttributes,
+  UserCreationAttributes,
+} from '../models/User';
+
+export default class AuthenticationController {
+  private userService: UserService;
+  private emailService: EmailService;
+
+  constructor(userService: UserService, emailService: EmailService) {
+    this.userService = userService;
+    this.emailService = emailService;
+  }
+
+  async signUp(req: Request, res: Response, next: NextFunction) {
+    try {
+      const {email, password, passwordConfirmation} = req.body;
+      const user = await this.userService.getOneUserByEmail(email);
+
+      if (user) {
+        res.status(400);
+        res.json({message: userFriendlyMessage.failure.emailExists});
+        return;
+      }
+      if (password !== passwordConfirmation) {
+        res.status(400);
+        res.json({
+          message: userFriendlyMessage.failure.passwordConfirmationMismatch,
+        });
+        return;
+      }
+
+      const createdUser = await this.userService.createOneUser({
+        ...req.body,
+        role: 'Student',
+        isConfirmed: false,
+        isTemporary: false,
+      });
+      const payload: Payload = {
+        id: createdUser.id,
+        role: createdUser.role,
+        schoolId: createdUser.schoolId,
+        isConfirmed: createdUser.isConfirmed,
+        isTemporary: createdUser.isTemporary,
+      };
+      const accessToken = JWTUtils.generateAccessToken(payload, undefined, {});
```

Comment:
Why is the token here undefined? Shouldn't it be the default accessToken? does putting undefined here force it to use the default param?

Reply:
This token should expire in 180 days

Reply:
Yes, putting it undefined will force it to use the default param. [Ref](https://www.typescripttutorial.net/typescript-tutorial/typescript-default-parameters/#:~:text=Use%20default%20parameter%20syntax%20parameter,the%20undefined%20into%20the%20function.)

But, I'll just make more methods in JWTUtils to generate access token for each scenario, for better clarity.